query/src/main/java/org/infinispan/query/impl/massindex/IndexWorker.java

```diff
@@ -162,14 +172,19 @@ public void writeObject(ObjectOutput output, IndexWorker worker) throws IOExcept
       @Override
       public IndexWorker readObject(ObjectInput input) throws IOException, ClassNotFoundException {
          String cacheName = (String) input.readObject();
-         Class indexedClass = (Class) input.readObject();
+         int typesSize = input.readInt();
+         Set<IndexedTypeIdentifier> types = new HashSet<>(typesSize);
```

Comment:
Is the initial capacity really helping ?

Reply:
TBH using or not initial capacity here is irrelevant: this Set will be generally small, and will have be created almost never